### PR TITLE
Leshikus/cuda

### PR DESCRIPTION
**Pull request by leshikus**

### CODE REVIEW

File: scripts/mapd-deps-conda-dev-env.yml [hunk 1]
```diff
@@ -55,6 +55,5 @@ dependencies:
   - ittapi
   - pytest
   # when cuda is enabled, install also
-  - nvcc_linux-64
```
Comment (Garra1980) [inline scripts/mapd-deps-conda-dev-env.yml:58]: why do we delete this?

Comment (leshikus) [reply scripts/mapd-deps-conda-dev-env.yml:58]: nvcc_linux-64 does not install correctly from conda-forge, cudatoolset-dev does

Comment (leshikus) [reply scripts/mapd-deps-conda-dev-env.yml:58]: cudatoolset-dev contains both nvcc and required dynamic libraries

File: scripts/conda/build.sh [hunk 2]
```diff
@@ -36,6 +36,13 @@ export RUN_TESTS=${RUN_TESTS:-0}
 
 export INSTALL_BASE=. # was opt/omnisci-cpu
 
+if [[ "$cuda_compiler_version" == None ]] || [[ -z "$cuda_compiler_version" ]]
+then
+   export EXTRA_CMAKE_OPTIONS="$EXTRA_CMAKE_OPTIONS -DENABLE_CUDA=off -DENABLE_DBE=on"
+else
+   export EXTRA_CMAKE_OPTIONS="$EXTRA_CMAKE_OPTIONS -DENABLE_CUDA=on -DENABLE_DBE=off"
```
[the remark this answers is not in the document]
Comment (leshikus) [reply scripts/conda/build.sh:43]: that's the point of this fix - either DBE or CUDA is enabled, not both
they do not work together anyway

File: scripts/mapd-deps-conda-dev-env.yml (hunk 1 above)
Comment (Garra1980) [reply scripts/mapd-deps-conda-dev-env.yml:58]: @kurapov-peter could you please also take a look

File: scripts/conda/build.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (Garra1980) [reply scripts/conda/build.sh:43]: ah, ok. DBE will be removed shortly anyway

File: scripts/mapd-deps-conda-dev-env.yml (hunk 1 above)
Comment (kurapov-peter) [reply scripts/mapd-deps-conda-dev-env.yml:58]: nvcc_linux-64 doesn't work for me either. I'll try the gpu build with the script this week

Comment (kurapov-peter) [reply scripts/mapd-deps-conda-dev-env.yml:58]: Complete list:
[  FAILED  ] 15 tests, listed below:
[  FAILED  ] Select.FilterAndSimpleAggregation
[  FAILED  ] Select.AggregateConstantValueOnEmptyTable
[  FAILED  ] Select.AggregateOnEmptyTable
[  FAILED  ] Select.FloatAndDoubleTests
[  FAILED  ] Select.InValues
[  FAILED  ] Select.FilterAndMultipleAggregation
[  FAILED  ] Select.ComplexQueries
[  FAILED  ] Select.Case
[  FAILED  ] Select.DivByZero
[  FAILED  ] Select.ReturnNullFromDivByZero
[  FAILED  ] Select.LogicalValues
[  FAILED  ] Select.DecimalCompression
[  FAILED  ] Select.Subqueries
[  FAILED  ] Select.Sample
[  FAILED  ] Select.LogicalSizedColumns

Comment (kurapov-peter) [reply scripts/mapd-deps-conda-dev-env.yml:58]: I assume this either relates to llvm (conda installs 9.0.1 for me) or arrow (ENABLE_ARROW_4=on) version.
conda list:
arrow-cpp                 5.0.0           py39h9266689_29_cuda    conda-forge
arrow-cpp-proc            3.0.0                      cuda    conda-forge

I'm going to try a newer llvm. See if that helps. @alexbaden any other ideas?

Comment (kurapov-peter) [reply scripts/mapd-deps-conda-dev-env.yml:58]: same result with llvm 11

Comment (kurapov-peter) [reply scripts/mapd-deps-conda-dev-env.yml:58]: updating drivers to 11.6 didn't help either

Comment (kurapov-peter) [reply scripts/mapd-deps-conda-dev-env.yml:58]: llvm 11 actually breaks arrow based tests btw

**Comment by kurapov-peter** [comment]
Mind if I merge this in?

**Comment by alexbaden** [comment]
Relates to #331

**Comment by kurapov-peter** [comment]
One more thing. It seems like conda installs boost-cpp version 1.74 with the script despite the default conda forge version is 1.78.
Any boost <=1.76 produces a bunch of annoying deprecation warnings (see i.e. https://github.com/boostorg/graph/pull/218). 
Is it the same for you?

**Comment by alexbaden** [comment]
I see the deprecation issues -- am looking into creating a conda package using `conda-forge` deps, or using `vcpkg` to manage deps (or both). 
I am also hoping landing #351 will help, even if it doesn't remove many dependencies, because it will likely remove some usages -- at least cross-module usages which makes dependencies even more complicated.

**Comment by kurapov-peter** [comment]
> I see the deprecation issues -- am looking into creating a conda package using `conda-forge` deps, or using `vcpkg` to manage deps (or both). I am also hoping landing #351 will help, even if it doesn't remove many dependencies, because it will likely remove some usages -- at least cross-module usages which makes dependencies even more complicated.

Yes, it does remove the problematic usage, but still. I was able to manually update boost version with `conda update`. Not sure why it's not installing the latest version right off.

**Comment by Garra1980** [comment]
> I see the deprecation issues -- am looking into creating a conda package using `conda-forge` deps, or using `vcpkg` to manage deps (or both). I am also hoping landing #351 will help, even if it doesn't remove many dependencies, because it will likely remove some usages -- at least cross-module usages which makes dependencies even more complicated.

#351 merged, can we proceed with this one?

**Pull request closed without merging**